src/osdc/Filer.cc

```diff
@@ -355,7 +355,8 @@ void Filer::_do_purge_range(PurgeRange *pr, int fin, int err)
 		 << dendl;
 
   if (pr->num == 0 && pr->uncommitted == 0) {
-    pr->oncommit->complete(pr->err);
+    if (pr->oncommit)
+      pr->oncommit->complete(pr->err);
```

Comment:
i am not familiar with Filer or MDLog. and i tried to search the caller where `oncommit` is null,  but failed. to be specific, i think `Journaler::erase()` is the caller of this method. but none of the paths passes `nullptr` to `Journaler::erase()` .

Reply:
Now that I look more closely, I don't see a way either for `oncommit` to be null. I'm just going to close this.